skxray/cdi.py

```diff
@@ -281,6 +282,12 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
     n_iterations : int, optional
         number of iterations to run.
         default is 1000.
+    cb_function : function, optional
+        This is a callback function that expects to receive these
+        four objects: sample_obj, obj_error, diff_error, sup_error
```

Comment:
You should also list the expected types.  Imaging reading this doc string with out having read the function, but wanting to write a callback to plug into this.